app/utils/EventReportingHelper.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import connectors.UserAnswersCacheConnector
+import identifiers.EventReportingId
+import models.{EventReporting, ListOfSchemes}
+import models.requests.AuthenticatedRequest
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Call
+import uk.gov.hmrc.http.{HeaderCarrier, SessionKeys}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object EventReportingHelper {
+
+  final case class EventReportingData(data:EventReporting, sessionStorageKey:String)
+  def eventReportingData(srn:String, listOfSchemes: ListOfSchemes, schemeName:String, returnUrl: Call)
```

Comment:
I'm not sure this method or this helper is really necessary - it seems to make things more not less complicated, though maybe that's just me. This code is only used twice + what it's doing is straightforward so I think this is a case where a small amount of repetition (i.e. only twice) of code is OK.